### DISCUSSION TITLE
Target VBUS

### DISCUSSION THREAD

**Discussion by phdussud** [ataradov/usb-sniffer#6] [Q&A] [answered]
It seems to me that the target VBUS isn't powered by the board. If this is correct, this makes the testing of bus powered devices much more complicated as one needs to patch VBUS of the device to power it. Can you suggest a good way of doing this? 
Thanks!

**Comment by ataradov** [comment]
This is just a sniffer, it does not interact with the bus at all, it just observes. It is essentially inserted inline with the device cable.  You have a host and a device connector. VBUS would be provided by the host, just like if there was no sniffer.

This is not a tool that can do anything with the device under test alone. You need to have the host connection.

**Reply by phdussud** [reply]
I get it. So you need to have a Y adapter of some sort. One connector goes to the host, one goes to the usb sniffer and one goes to the device.

**Comment by ataradov** [comment, marked as answer]
You need two host USB ports - one if for the sniffed data and one for the actual device under test. Those may not even be on the same PC. If they are on the same PC, then it is better to connect them to the root hub (ports on the PC case), not to an external hub. Especially if you want to do USB HS sniffing. Any additional hubs limit the available bandwidth.

And then you have USB-A connector for the device under test, of course.

**Reply by phdussud** [reply]
Right! Thanks!
